tests/test_encoding/test_mean_encoder.py

```diff
@@ -126,11 +126,19 @@ def test_warning_if_transform_df_contains_categories_not_present_in_fit_df(
 ):
     # test case 4: when dataset to be transformed contains categories not present
     # in training dataset
-    with pytest.warns(UserWarning):
+
+    msg = "During the encoding, NaN values were introduced in the feature(s) var_A."
+
+    with pytest.warns(UserWarning) as record:
         encoder = MeanEncoder()
```

Comment:
specify inside the transformer rare_labels='ignore' to make clear what we are testing